ui/src/app/components/side-nav.tsx

```diff
@@ -88,293 +145,229 @@ interface SideNavItemProps {
-                containsSubItems={true}
-                ref={this.state.adminRef}
-        />
-      }
-      {
-        hasAuthorityForAction(profile, AuthorityGuardedAction.USER_ADMIN) && this.state.showAdminOptions && <SideNavItem
+        onToggleSideNav={() => props.onToggleSideNav()}
+        parentOnClick={() => openContactWidget()}
+    />
+    {hasAuthorityForAction(props.profile, AuthorityGuardedAction.SHOW_ADMIN_MENU) && <SideNavItem
+        icon='user'
+        content='Admin'
+        parentOnClick={() => onToggleAdmin()}
+        onToggleSideNav={() => props.onToggleSideNav()}
+        containsSubItems={true}
+    />
+    }
+    {
+      hasAuthorityForAction(props.profile, AuthorityGuardedAction.USER_ADMIN) && showAdminOptions && <SideNavItem
           content={'User Admin'}
-          onToggleSideNav={() => this.props.onToggleSideNav()}
+          onToggleSideNav={() => props.onToggleSideNav()}
           href={'/admin/user'}
-          active={this.props.userAdminActive}
-        />
-      }
-      {
-        hasAuthorityForAction(profile, AuthorityGuardedAction.USER_AUDIT) && this.state.showAdminOptions && <SideNavItem
-            content={'User Audit'}
-            onToggleSideNav={() => this.props.onToggleSideNav()}
-            href={'/admin/user-audit/'}
-            active={this.props.userAuditActive}
-        />
-      }
-      {
-        hasAuthorityForAction(profile, AuthorityGuardedAction.SERVICE_BANNER) && this.state.showAdminOptions && <SideNavItem
-            content={'Service Banners'}
-            onToggleSideNav={() => this.props.onToggleSideNav()}
-            href={'/admin/banner'}
-            active={this.props.bannerAdminActive}
-        />
-      }
-      {
-        hasAuthorityForAction(profile, AuthorityGuardedAction.WORKSPACE_ADMIN) && this.state.showAdminOptions && <SideNavItem
-            content={'Workspaces'}
-            onToggleSideNav={() => this.props.onToggleSideNav()}
-            href={'admin/workspaces'}
-            active={this.props.workspaceAdminActive}
-        />
-      }
-      {
-        hasAuthorityForAction(profile, AuthorityGuardedAction.WORKSPACE_AUDIT) && this.state.showAdminOptions && <SideNavItem
-            content={'Workspace Audit'}
-            onToggleSideNav={() => this.props.onToggleSideNav()}
-            href={'/admin/workspace-audit/'}
-            active={this.props.workspaceAuditActive}
-        />
-      }
-      {
-        hasAuthorityForAction(profile, AuthorityGuardedAction.INSTITUTION_ADMIN) && this.state.showAdminOptions && <SideNavItem
-            content={'Institution Admin'}
-            onToggleSideNav={() => this.props.onToggleSideNav()}
-            href={'admin/institution'}
-            active={this.props.workspaceAdminActive}
-        />
-      }
-    </div>;
-  }
-}
+          active={userAdminActive()}
+      />
+    }
+    {
+      hasAuthorityForAction(props.profile, AuthorityGuardedAction.USER_AUDIT) && showAdminOptions && <SideNavItem
```

Comment:
nit: hasAuthorityForAction(props.profile, AuthorityGuardedAction.USER_AUDIT) && showAdminOptions && <SideNavItem this could be a function with arguments: authority, content, href and active function

Reply:
I may do something like this next, to cut down on the amount of boilerplate, but not in this PR.